lib/adapter.ts

```diff
@@ -163,6 +163,12 @@ export default class ObjectionAdapter extends ORMAdapter {
     }
   }
 
+  foreignKeyForRelationship(model: DenaliModel): string {
```

Comment:
I'm thinking we don't want to base this off the model, but rather the relationship. You might have a model with two relationships to another model, i.e.`Book` which has many `authors` as well as `editors`, but both of those are relationships to `User`.

So I'd expect this argument to be `relationshipName`, or even better, the full Denali `HasManyRelationship` or `HasOneRelationship` descriptor object.

Reply:
How would you get the type then? What are you thinking it would look like exactly?

Reply:
> How would you get the type then? What are you thinking it would look like exactly?
  
Perhaps I'm misunderstanding your question, but if you have the descriptor for the relationship, [that contains the type](https://github.com/denali-js/denali/blob/master/lib/data/descriptors.ts#L22).

So the implementation would look very similar to your current PR implementation, except the argument would be the descriptor for the relationship.

Reply:
That makes sense about the descriptor.

The `to` and `from` is coming from 
https://github.com/seawatts/denali-objection/blob/master/lib/relation-mappings/many-to-many.ts#L48

We only have one relationship descriptor here. So what would we pass in?

Reply:
So many-to-many is a bit of a unique case because you have a join table with _two_ foreign keys for the same relationship (i.e. the `post_tags` table with `post_id` and `tag_id` columns). All other kinds of relationships, you'd have one model with an `id` and the other model with a foreign key (`<relationship name>_id`).

Now in the case of a user-defined join model (i.e. a `post_tags` table with additional data like `tag_added_at`), the user would create their own Denali model and use regular one-to-many relationships to setup the many-to-many (i.e. `class PostTag extends Model {  post: hasOne(), tag: hasOne() }`). So that scenario won't hit this branch of code - we can forget about it for now.

The only time (for Objection at least) that a many-to-many is used is when the user effectively wants the ORM to manage the join table for them (an "implicit" join table). 

Also, remember that Denali's relationship descriptors only describe _one side_ of the relationship. `hasMany` and `hasOne`, not `oneToMany` and `oneToOne`. That means that the many-to-many relationship must either be explicitly flagged as such by the user (`hasMany('tag', { manyToMany: true })`), or must be inferred (Post `hasMany('tag')` and Tag `hasMany('post')` - must be many-to-many). The latter is nicer for the user, but complicated (what if the models have multiple relationships? Now we need to compare relationship names to find the other side of the relationship ...).

So for the first pass, I decided to go the explicit route - you have to tell the Objection Adapter that you want this to be a many-to-many relationship:

```
class Post extends Model {
  tag: hasMany('tag', { manyToMany: true })
```

With the above option, denali-objection will hit [this line](https://github.com/seawatts/denali-objection/blob/master/lib/define-models.ts#L67), causing it to define a many-to-many relationship. Notice that the function on that line is taking a model and mapping over all it's relationship descriptors. That's where you are getting the descriptor from.

So I see two options here:

1. In the many-to-many relation mapper, we look up the other side of the relationship and use _it's_ descriptor to generate the foreign key for the other foreign key column in the join table. This would also let us drop the explicit `manyToMany` option requirement.
2. We add another config option that denali-objection will read from `hasMany` that defines the column names for the foreign keys in the join tables (and thus, many-to-many relationships would effectively ignore the `foreignKeyForRelationship` hook.

Another option I considered, but rejected, was to pass the model into `foreignKeyForRelationship` (i.e. we have Model A, and the descriptor that points to Model B, so we'd change `foreignKeyForRelationship` to take a descriptor or a model and work from both). But ultimately that's an ugly solution (mixed argument types is confusing), and I'd rather not set the precedent that `foreignKeyForRelationship` takes a mixed argument (making it harder to change to the ideal solution - inferring the other side of the relationship - down the line).